### PR TITLE
Shorten tight_bbox.

### PR DESCRIPTION
**Pull request by anntzer**
It's more readable to directly define the relevant callbacks
(axes_locator, apply_aspect) as one-line lambdas at the point of use,
instead of having to define them much earlier.  Pickling is not an issue
as these lambdas are only set temporarily while computing the figure
tightbbox for saving.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/tight_bbox.py [hunk 1]
```diff
@@ -18,33 +18,23 @@ def adjust_bbox(fig, bbox_inches, fixed_dpi=None):
...
-        stack.enter_context(_setattr_cm(ax, apply_aspect=no_op_apply_aspect))
-
-    if fixed_dpi is not None:
-        tr = Affine2D().scale(fixed_dpi)
-        dpi_scale = fixed_dpi / fig.dpi
-    else:
-        tr = Affine2D().scale(fig.dpi)
-        dpi_scale = 1.
+        ax.set_axes_locator(
+            lambda a, r, _pos=ax.get_position(original=False).frozen(): _pos)
```
Comment (jklymak) [inline lib/matplotlib/tight_bbox.py:30]: I disagree that this is more readable....

Comment (anntzer) [reply lib/matplotlib/tight_bbox.py:30]: went for some compromise :)

**Comment by QuLogic** [comment]
Is the docstring okay instead of a comment? Not sure what part you're referring to.

**Comment by anntzer** [comment]
I guess @jklymak is hoping for more comments, but let's leave that for another time :)

**Comment by jklymak** [comment]
I find 
```python
         _pos = ax.get_position(original=False).frozen()
         ax.set_axes_locator(lambda a, r, _pos=_pos: _pos)
```
to be pretty obscure and self-referential...

**Comment by anntzer** [comment]
It's basically the same as before the PR?

File: lib/matplotlib/tight_bbox.py [hunk 2]
```diff
@@ -18,33 +18,23 @@ def adjust_bbox(fig, bbox_inches, fixed_dpi=None):
...
-        stack.enter_context(_setattr_cm(ax, apply_aspect=no_op_apply_aspect))
-
-    if fixed_dpi is not None:
-        tr = Affine2D().scale(fixed_dpi)
-        dpi_scale = fixed_dpi / fig.dpi
-    else:
-        tr = Affine2D().scale(fig.dpi)
-        dpi_scale = 1.
+        _pos = ax.get_position(original=False).frozen()
+        ax.set_axes_locator(lambda a, r, _pos=_pos: _pos)
```
Comment (jklymak) [inline lib/matplotlib/tight_bbox.py:30]:
```suggestion
        ax.set_axes_locator(lambda a, r, pos=_pos: pos)
```
would be clearer to me...

Comment (anntzer) [reply lib/matplotlib/tight_bbox.py]: went for a better name

Comment (jklymak) [reply lib/matplotlib/tight_bbox.py]: Thanks that's much clearer.  I have enough of a mental block about lambdas without the kwargs having the same name as their arguments 😃

**Pull request merged**